clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -3271,12 +3271,48 @@ static void handleWeakImportAttr(Sema &S, Decl *D, const ParsedAttr &AL) {
   D->addAttr(::new (S.Context) WeakImportAttr(S.Context, AL));
 }
 
+// Handles reqd_work_group_size and work_group_size_hint.
+template <typename WorkGroupAttr>
+static void handleWorkGroupSize(Sema &S, Decl *D, const ParsedAttr &AL) {
+  uint32_t WGSize[3];
+  for (unsigned i = 0; i < 3; ++i) {
+    const Expr *E = AL.getArgAsExpr(i);
+    if (!checkUInt32Argument(S, AL, E, WGSize[i], i,
+                             /*StrictlyUnsigned=*/true))
+      return;
+    if (WGSize[i] == 0) {
+      S.Diag(AL.getLoc(), diag::err_attribute_argument_is_zero)
+          << AL << E->getSourceRange();
+      return;
+    }
+  }
+
+  WorkGroupAttr *Existing = D->getAttr<WorkGroupAttr>();
+  if (Existing && !(Existing->getXDim() == WGSize[0] &&
+                    Existing->getYDim() == WGSize[1] &&
+                    Existing->getZDim() == WGSize[2]))
+    S.Diag(AL.getLoc(), diag::warn_duplicate_attribute) << AL;
+
+  D->addAttr(::new (S.Context)
+                 WorkGroupAttr(S.Context, AL, WGSize[0], WGSize[1], WGSize[2]));
+}
+
```

Comment:
@steffenlarsen, could you please explain why you added this function? I do not see any tests regarding this change. Are you planning to remove the existing support for work_group_size_hint attribute?

Reply:
This comes from upstream. Since we now have `reqd_work_group_size` in both a SYCL and an upstream-like variant I reverted as much as possible back to upstream. I expect to give `work_group_size_hint` the same treatment once we agree on this approach.

Reply:
Thanks for the explanation!

Reply:
>>Since we now have reqd_work_group_size in both a SYCL and an upstream-like variant I reverted as much as possible back to upstream. I expect to give work_group_size_hint the same treatment once we agree on this approach.

@steffenlarsen, could you please create an issue about it so that we do not forget this discussion? you can assign to me as needed.

Reply:
https://github.com/intel/llvm/issues/7699